src/node/audio_buffer_source.rs

```diff
@@ -75,8 +77,8 @@ pub struct AudioBufferSourceNode {
     sender: Sender<AudioBufferMessage>,
     detune: AudioParam,        // has constraints, no a-rate
     playback_rate: AudioParam, // has constraints, no a-rate
-    buffer: Option<AudioBuffer>,
-    source_started: bool,
+    buffer: OnceCell<AudioBuffer>,
+    source_started: AtomicBool,
```

Comment:
Don't understand why we need an `AtomicBool` here?

Reply:
It's because the following function can now be used in a multithreaded case (takes an `&self` instead of `&mut self`)
```
    pub fn start_at_with_offset_and_duration(&self, start: f64, offset: f64, duration: f64) {
        if self.source_started.swap(true, Ordering::SeqCst) {
            panic!("InvalidStateError: Cannot call `start` twice");
        }
     ...
```

I can't change the bool value if I only have `&self`, but I can change an AtomicBool!
https://doc.rust-lang.org/std/sync/atomic/struct.AtomicBool.html#method.swap
As you can see the atomicbool is mutable without needing `&mut self` because it leverages atomic CPU instructions.

The reason I would like to have all AudioNode methods take `&self` instead of `&mut self` is, as you stated, user friendlyness (it "just works" if you call from different threads) and also resembles the spec better

Reply:
oh ok thanks! (I somehow need to update my mindset regarding threads :)